core/patina_debugger/src/debugger.rs

```diff
@@ -211,10 +216,15 @@ impl<T: SerialIO> PatinaDebugger<T> {
 
         let mut timeout_reached = false;
         if let GdbStubStateMachine::Idle(mut inner) = gdb {
-            // Always start with a stop code if starting from idle. This may be because this is the initial breakpoint
+            // If this is a restart, send a nack to request a resend of the failing packet.
+            // Otherwise, always start with a stop code if starting from idle. This may be because this is the initial breakpoint
             // or because the initial breakpoint timed out. This is not to spec, but is a useful hint to the client
             // that a break has occurred. This allows the debugger to reconnect on scenarios like reboots.
-            let _ = inner.borrow_conn().write_all(GDB_STOP_PACKET.as_bytes());
+            if restart {
+                self.transport.write(GDB_NACK_PACKET);
```

Comment:
Why `self.transport.write` here vs the `inner.borrow_con.write_all()` for the other one?